egi2bids/mff2bids.py

```diff
@@ -44,28 +50,71 @@
     "249", "250", "251", "F9", "253", "254", "255", "256",
     "Cz",
 ]
+
+
 # fmt: on
+def _extract_folder(file):
+    """Extracts a compressed folder to its original form"""
+    # Get the path and file extension
+    file = Path(file)
+    ext = file.suffix
+    # Open the archive file based on its type
+    print(file)
+    if ext == ".tar":
+        path = file.with_name(file.name.replace(".tar", ""))
+        logger.info(f"Extracting zip archive to {file} to {path}")
+        archive = tarfile.open(file)
+        archive.extractall(path)
+        archive.close()
+
+    elif ext == ".gz":
+        path = file.with_name(file.name.replace(".tar.gz", ""))
+        logger.info(f"Extracting zip archive to {file} to {path}")
+        archive = tarfile.open(file, "r:gz")
+        archive.extractall(path)
+        archive.close()
+
+    elif ext == ".zip":
+        path = file.with_name(file.name.replace(".zip", ""))
+        logger.info(f"Extracting zip archive to {file} to {path}")
+        archive = zipfile.ZipFile(file)
+        archive.extractall(path)
+        archive.close()
+
+    elif ext == ".mff":
+        path = file
+    else:
+        raise ValueError(f"Unsupported file type: {ext}")
+    print(path)
+    return path
 
 
 def mff2bids(
     mff_source,
     bids_root,
-    subject=None,
-    task=None,
-    session=None,
+    subject,
+    session,
+    task,
     run=None,
     event_id=None,
     line_frequency=50,
+    save_source=False,
+    overwrite=False,
 ):
-    bids_path = BIDSPath(root=Path(bids_root))
+    # mff_source
+    logger.info(mff_source)
+    # Extract
+    mff_source = _extract_folder(mff_source)
+
+    # BIDS path
+    bids_root = Path(bids_root)
+    bids_path = BIDSPath(root=bids_root)
```

Comment:
The update below can be done in a single line providing all arguments to the same call to `.update`, no?